### PR TITLE
Refactor dev-env variable names

### PR DESCRIPTION
**Pull request by Rozzii**
The goal of this commit is to make the variable names involved in
network configuration more explicit. The main goal of this commit is to
make the difference between the "provisioner" network and the "provisioning"
network more prononced.

In the future the dev-env will support at least 2 different
network topologies thus the provisioner (e.g. ironic) won't be necessarily available
in the provisioning network the same way as it is available now.

This commit is a forruner of future changes that will introduce additional network
organization options.

List of name changes:

 - CLUSTER_PROVISIONING_INTERFACE -> BARE_METAL_PROVISIONER_INTERFACE
 - PROVISIONING_IP -> BARE_METAL_PROVISIONER_IP
 - PROVISIONING_IPV6 -> BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY
 - PROVISIONING_NETWORK -> BARE_METAL_PROVISIONER_NETWORK
 - PROVISIONING_CIDR -> BARE_METAL_PROVISIONER_CIDR
 - PROVISIONING_NETMASK -> BARE_METAL_PROVISIONER_NETMASK
 - CLUSTER_PROVISIONING_IP -> CLUSTER_BM_PROVISIONER_IP
 - PROVISIONING_URL_HOST -> BARE_METAL_RPOVISIONER_URL_HOST
 - CLUSTER_URL_HOST -> CLUSTER_BARE_METAL_PROVISIONER_HOST
 - VIRSH_DHCP_V4_START -> EXTERNAL_DHCP_V4_START
 - VIRSH_DHCP_V4_END -> EXTERNAL_DHCP_V4_START
 - BAREMETALV4_POOL_RANGE_START -> IPAM_EXTERNALV4_POOL_RANGE_START
 - BAREMETALV4_POOL_RANGE_END -> IPAM_EXTERNALV4_POOL_RANGE_END
 - VIRSH_DHCP_V6_START ->  EXTERNAL_DHCP_V6_START
 - VIRSH_DHCP_V6_END -> EXTERNAL_DHCP_V6_END
 - BAREMETALV6_POOL_RANGE_START ->  IPAM_EXTERNALV6_POOL_RANGE_START
 - BAREMETALV6_POOL_RANGE_END -> IPAM_EXTERNALV6_POOL_RANGE_END
 - INITIAL_IRONICBRIDGE_IP -> INITIAL_BM_PROVISIONER_BRIDGE_IP
 - baremetal_network_cidr_v4 -> external_network_cidr_v4
 - baremetal_network_cidr_v6 -> external_network_cidr_v6
 - virsh_dhcp_v4_start -> external_dhcp_v4_start
 - virsh_dhcp_v4_end -> external_dhcp_v4_end
 - virsh_dhcp_v6_start -> external_dhcp_v6_start
 - virsh_dhcp_v6_end -> external_dhcp_v6_end
 - PROVISIONING_POOL_RANGE_START -> IPAM_PROVISIONING_POOL_RANGE_START
 - PROVISIONING_POOL_RANGE_END -> IPAM_PROVISIONING_POOL_RANGE_END
 - MANAGE_BR_BRIDGE -> MANAGE_EX_BRIDGE
 - every variable that had BM_ in it's name has BARE_METAL_ instead now

### CODE REVIEW

**Comment by Rozzii** [comment]
/hold 
I am testing, it is not done yet.

**Comment by mboukhalfa** [comment]
testing the new trigger format
/test-centos-e2e-integration-main

File: 04_verify.sh
Comment (tuminoid) [inline 04_verify.sh:25]: Rather than making it an extra long line, just have two lines of `local ...`

File: 04_verify.sh
Comment (tuminoid) [inline 04_verify.sh:37]: Curlys missing.

File: 04_verify.sh
Comment (tuminoid) [inline 04_verify.sh:81]: Curlys missing.

File: host_cleanup.sh
Comment (tuminoid) [inline host_cleanup.sh:62]: `==` -> `=`

File: lib/images.sh
Comment (tuminoid) [inline lib/images.sh:27]: Curlys missing, and quote the right hand side.

File: lib/images.sh
Comment (tuminoid) [inline lib/images.sh:38]: Same.

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:33]: Fix quotes to go around the assignment instead inside the braces.

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:42]: `==` -> `=`

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:44]: Add quotes.

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:47]: Same.

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:64]: This has so many ways to fail and slip through, can we unpack it into an `if` statement please.

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:66]: Entire lib is missing curlys, so I'm not going to complain about those.

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:73]: `==` -> `=`

File: lib/network.sh
Comment (tuminoid) [inline lib/network.sh:196]: Move quotes around the whole assignment. Same for all lines below.

File: tests/roles/run_tests/templates/clusterctl-vars.yaml
Comment (tuminoid) [inline tests/roles/run_tests/templates/clusterctl-vars.yaml:20]: Is the `BARE_METAL` vs `BAREMETAL` inconsistency something we want to have?

File: ubuntu_bridge_network_configuration.sh
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:23]: `==` -> `=`

File: ubuntu_bridge_network_configuration.sh
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:24]: Curlys.

File: ubuntu_bridge_network_configuration.sh
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:26]: Curlys.

File: ubuntu_bridge_network_configuration.sh
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:38]: `==` -> `=`

File: ubuntu_bridge_network_configuration.sh
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:40]: What is this supposed to be doing? It produces output but has no test.

Probably going for `if ! ip a show external &>/dev/null; then`  as a way to check if `external` exists?

File: ubuntu_bridge_network_configuration.sh
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:61]: `==` -> `=`

**Review by tuminoid — COMMENTED** [review]
Some nits.

File: lib/images.sh
Comment (Rozzii) [reply lib/images.sh:38]: done

File: 02_configure_host.sh
Comment (Rozzii) [reply 02_configure_host.sh:63]: done

File: 04_verify.sh
Comment (Rozzii) [reply 04_verify.sh:25]: done

File: 04_verify.sh
Comment (Rozzii) [reply 04_verify.sh:37]: done

File: 04_verify.sh
Comment (Rozzii) [reply 04_verify.sh:74]: done

File: 04_verify.sh
Comment (Rozzii) [reply 04_verify.sh:81]: done

File: host_cleanup.sh
Comment (Rozzii) [reply host_cleanup.sh:62]: done

File: lib/images.sh
Comment (Rozzii) [reply lib/images.sh:27]: done

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh:33]: done

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh:42]: done

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh:47]: done

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh:73]: done

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh:196]: done

File: ubuntu_bridge_network_configuration.sh
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:24]: done

File: ubuntu_bridge_network_configuration.sh
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:26]: done

File: ubuntu_bridge_network_configuration.sh
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:38]: done

File: ubuntu_bridge_network_configuration.sh
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:61]: done

File: ubuntu_bridge_network_configuration.sh
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh]: done

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh]: done

File: tests/roles/run_tests/templates/clusterctl-vars.yaml
Comment (Rozzii) [reply tests/roles/run_tests/templates/clusterctl-vars.yaml:20]: done, i have changed every BAREMETAL to BARE_METAL.

File: ubuntu_bridge_network_configuration.sh
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:40]: yeah I think the original purpose of this condition was to check whether the external interface is present or not.

**Review by tuminoid — COMMENTED** [review]
Just more nits. I think these need cleanup in separate patch as all convention is completely random, and there are so many bad practices visible in these snippets. So feel free to fix/resolve any you feel like. No actual problems spotted with renaming.

File: lib/network.sh
Comment (Rozzii) [reply lib/network.sh:1]: I will fix curling and = signs in a separate refactoring PR I think most of the files in dev env have these incorrectly so I would prefer pushing a dedicated syntax fixing PR

Comment (tuminoid) [reply lib/network.sh:1]: That works for me. Thanks @Rozzii

**Comment by Rozzii** [comment]
Followup PR will implement shell syntax improvements mentioned by comments here.

**Review by lentzi90 — CHANGES_REQUESTED** [review]
Overall good improvements! It is now clearer what these variables do (but I must confess I still don't understand all of them :sweat_smile: ).

I'm a bit unsure if it is necessary to have `BARE_METAL` in all of them? For example, are there other provisioners than bare metal provisioners? Or could we just say `PROVISIONER` then instead of `BARE_METAL_PROVISIONER`? Just a thought :slightly_smiling_face: 

Upgrade vars should not be added back (see below)!

File: tests/feature_tests/upgrade/upgrade_vars.sh
Comment (Rozzii) [reply tests/feature_tests/upgrade/upgrade_vars.sh:1]: Thanks fort detecting this

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by tuminoid — DISMISSED** [review]
As agreed, my nits are addressed in separate PRs. There are some other pending changes so just signalling my approval for now.

File: vm-setup/roles/firewall/tasks/iptables.yaml
Comment (mboukhalfa) [inline vm-setup/roles/firewall/tasks/iptables.yaml:113]: We are still keeping provisioning network name ?

File: vm-setup/library/generate_macs.py
Comment (mboukhalfa) [inline vm-setup/library/generate_macs.py:33]: to me macs are not related to external network would prefer to name it something related to VM `generate_baremetal_vm_macs`

File: vm-setup/library/generate_macs.py
Comment (mboukhalfa) [inline vm-setup/library/generate_macs.py:39]: `external VMs` does not really make sense

File: vm-setup/library/generate_macs.py
Comment (mboukhalfa) [inline vm-setup/library/generate_macs.py:84]: check function definition comment

**Review by mboukhalfa — COMMENTED** [review]
few notes

File: vm-setup/roles/firewall/tasks/iptables.yaml
Comment (Rozzii) [reply vm-setup/roles/firewall/tasks/iptables.yaml:113]: Yes, eventually there will be Provisioner and  Provisioning networks.  But in this specifc case of "{{ provisioning_host_ports }}" I have to check I might have missed this.

File: vm-setup/library/generate_macs.py
Comment (Rozzii) [reply vm-setup/library/generate_macs.py]: done

File: vm-setup/library/generate_macs.py
Comment (Rozzii) [reply vm-setup/library/generate_macs.py]: done

File: vm-setup/roles/firewall/tasks/iptables.yaml
Comment (Rozzii) [reply vm-setup/roles/firewall/tasks/iptables.yaml:113]: I have changed "provisioning_host_ports" to "vm_host_ports"

**Review by tuminoid — APPROVED** [review]
/lgtm

**Comment by Rozzii** [comment]
Had to rebase looks like for some reason it was unable to get merged.
/test-centos-integration-release-main
/test-centos-integration-release-1-2
/test-ubuntu-e2e-integration-main
/test-centos-integration-release-1-3

**Pull request merged**
